src/util/taproot.rs

```diff
@@ -131,6 +131,21 @@ impl TapLeafHash {
     }
 }
 
+impl TapBranchHash {
+    /// Computes branch hash given two hashes of the nodes underneath it.
+    pub fn from_node_hashes(a: sha256::Hash, b: sha256::Hash) -> TapBranchHash {
```

Comment:
This is unsafe, we have a taproot hashjes type system(LeafHash/NodeHash) for a reason. This code can be mis-used. As such I don't see any reason why this code needs to be public.

Reply:
There are three main hashes in taptree hash type system:
- `TapLeafHash`, which is for the script leaf,
- `TapBranchHash`, which is for the script branch,
- `sha256::Hash`, when we do not known what it is (branch or a leaf, i.e. hidden nodes).

I am following that convention here. And of course, we can have a type alias for `sha256::Hash`, like `TapNodeHash`.

But I do not see how this breaks the existing type system in taproot module.